### DISCUSSION TITLE
Why ffmpeg is required?

### DISCUSSION THREAD

**Discussion by bemble** [blakeblackshear/frigate#4090] [General]
Hi,

Just a stupid question but why ffmpeg is required? I would like to use frigate just to get presence sensors in HA. I have a Coral, 5 cameras, and frigate makes my Pi4/8Gb unusable. On an old computer, it also uses a lot of CPU just for ffmpeg.

I'm just wondering why ffmpeg is required. I looked at the code and opencv2 is used which can take streams as input and extract the images from the streams.

Does anyone has the answer?

**Comment by NickM-27** [comment]
OpenCV does have support for reading streams, but it uses ffmpeg behind the scenes to do this anyway. 

Point is, at the end of the day something is needed to decode the streams into individual frames and that is a lot of work regardless of what software does it. It is also better to just use ffmpeg ourselves as we can control the behavior and we also do a lot of other manipulation to the frames to achieve optimal results for running motion and object detection.

If you are seeing high CPU usage then recommend reading https://github.com/blakeblackshear/frigate/issues/3860

**Reply by bemble** [reply]
Thanks for the explaination!
